### PR TITLE
Feature - Add service detail task list

### PR DESCRIPTION
**Pull request by orlandohohmeier**
task-list
Add a service detail task list which either displays a list of framework tasks including the scheduler tasks or a list of Marathon application tasks.

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @aldipower

**Comment by aldipower** [comment]
Yipii, starting reviewing.

File: src/js/components/FilterButtons.js [hunk 1]
```diff
@@ -36,8 +36,8 @@ class FilterButtons extends React.Component {
     let filterCount = this.getCount(itemList);
 
     return filters.map((filter) => {
-      let classSet = classNames({
-        'button button-stroke button-inverse': true,
+      let classSet = classNames('button button-stroke',{
```
Comment (kennyt) [inline src/js/components/FilterButtons.js]: can we add another `prop` to this component called `className` and have it default to `button button-stroke` and then use it here? that way, we can have the buttons not stroke if don't want it to :)

Comment (aldipower) [inline src/js/components/FilterButtons.js]: This is a good idea.
I suggest to add this only when needed. 
Could we make usage of this generalizing thing already?

File: src/js/components/FilterButtons.js [hunk 2]
```diff
@@ -62,19 +62,21 @@ class FilterButtons extends React.Component {
 }
 
 FilterButtons.defaultProps = {
-  renderButtonContent: function (title) {return title; },
-  onFilterChange: function () {}
+  inverseStyle: false,
+  onFilterChange: function () {},
+  renderButtonContent: function (title) {return title; }
```
Comment (aldipower) [inline src/js/components/FilterButtons.js]: Please add an extra space before the `return` and make sure your editors linting is working properly. Thank you.

File: src/js/components/FilterButtons.js [hunk 3]
```diff
@@ -62,19 +62,21 @@ class FilterButtons extends React.Component {
...
-  renderButtonContent: React.PropTypes.func,
   filters: React.PropTypes.array,
   // The key in itemList that is being filtered
   filterByKey: React.PropTypes.string,
+  inverseStyle: React.PropTypes.bool,
+  itemList: React.PropTypes.array.isRequired,
   // A function that returns the onClick for a filter button given the filter.
   onFilterChange: React.PropTypes.func,
-  itemList: React.PropTypes.array.isRequired,
+  // Optional function to generate button text. args: (filter, count)
```
Comment (aldipower) [inline src/js/components/FilterButtons.js]: I know it's not you comment. 

I don't trust the `args: (filter, count)`. Could you just remove this portion of the comment?

File: src/js/components/ServiceDetail.js [hunk 4]
```diff
@@ -48,7 +50,13 @@ class ServiceDetail
   }
 
   renderTasksTabView() {
-    return (<span>Tasks Placeholder</span>);
+    let {service} = this.props;
```
Comment (aldipower) [inline src/js/components/ServiceDetail.js]: `service` is only used once. No need to cache this.

File: src/js/components/ServiceDetail.js [hunk 5]
```diff
@@ -67,13 +75,22 @@ class ServiceDetail
             {this.tabs_getUnroutedTabs()}
           </ul>
         </div>
-        {this.tabs_getTabView()}
+        <div className="
```
Comment (aldipower) [inline src/js/components/ServiceDetail.js]: Not sure if the line breaks are correct here?
Are they?

The dangling `"`  looks strange.

File: src/js/components/FilterButtons.js (hunk 2 above)
Comment (orlandohohmeier) [inline src/js/components/FilterButtons.js]: My linting is working like charm - eslint just don't complains about the missing space. :(

File: src/js/components/TaskView.js [hunk 6]
```diff
@@ -133,55 +137,74 @@ class TaskView extends mixin(SaveStateMixin) {
...
+    let totalNumberOfTasks = tasks.length;
 
-    if (state.searchString !== '') {
-      tasks = StringUtil.filterByString(tasks, 'name', state.searchString);
+    // Get task states based on TaskStates types
+    let taskStates = tasks.map(function (task) {
+      let {stateTypes} = TaskStates[task.state];
+
+      return stateTypes.find(function (state) {
+        return state === 'active' || state === 'completed';
```
Comment (aldipower) [inline src/js/components/TaskView.js]: Please use the constant instead of the strings to compare.

TaksState.ACTIVE

File: src/js/components/FilterButtons.js (hunk 2 above)
Comment (kennyt) [inline src/js/components/FilterButtons.js]: i actually don't think we have a linting for that rule. there are specific cases where we don't do it. i think this is good 👍

File: src/js/components/TaskView.js (hunk 6 above)
Comment (orlandohohmeier) [inline src/js/components/TaskView.js]: There is no such thing as `TaksState.ACTIVE` – I also wanted to use the constants directly, and I think we should refactor the TaksState structure, but that should be part of a different PR.

File: src/js/components/ServiceDetail.js (hunk 4 above)
Comment (orlandohohmeier) [inline src/js/components/ServiceDetail.js]: There is in general no need to cache those objects, as the performance benefits are in the mini micro scale, but that's a different story.

File: src/js/components/FilterButtons.js (hunk 3 above)
Comment (orlandohohmeier) [inline src/js/components/FilterButtons.js]: I'm ok with removing it, but would keep it, as the documentation is valuable if you're implement your own `renderButtonContent` method. What do you think?

File: src/js/components/FilterButtons.js (hunk 2 above)
Comment (orlandohohmeier) [inline src/js/components/FilterButtons.js]: @kennyt You wouldn't add a `space`?

File: src/js/stores/MesosStateStore.js [hunk 7]
```diff
@@ -164,6 +164,37 @@ var MesosStateStore = Store.createStore({
     return [];
   },
 
+  getTasksByServiceId: function (serviceId) {
+    // Convert serviceId to Mesos service name
+    let mesosServiceName = serviceId.split('/').slice(1).reverse().join('.');
```
Comment (aldipower) [inline src/js/stores/MesosStateStore.js]: Yummy, spaghetti. Like this one here. :)

File: src/js/components/FilterButtons.js (hunk 3 above)
Comment (aldipower) [inline src/js/components/FilterButtons.js]: OK, fine with it. It just feels like the wrong place to document this.

File: src/js/components/TaskView.js (hunk 6 above)
Comment (aldipower) [inline src/js/components/TaskView.js]: OK sorry. I didn't had the structure of TaskStates in mind.
You're right, TaskStates should be refactored.
There should be constants created for this, but this is not part of this PR of course.

**Comment by aldipower** [comment]
Works like expected in FF. Awesome job. Thank you.

**Comment by aldipower** [comment]
Oh, I have linting errors while trying to run the tests. So I cannot run the tests.

**Comment by orlandohohmeier** [comment]
@aldipower I've rebased the branch and you shouldn't see linting errors anymore.

**Comment by aldipower** [comment]
Perfect! Thanx a lot!

**Comment by aldipower** [comment]
LGTM @Poltergeist Merge if you like.

File: src/js/components/ServiceDetail.js [hunk 8]
```diff
@@ -48,32 +50,43 @@ class ServiceDetail
...
-          <ServiceInfo service={service} />
+          <ServiceInfo service={this.props.service} />
           <ul className="tabs list-inline flush-bottom container-pod
             container-pod-short-top inverse">
             {this.tabs_getUnroutedTabs()}
           </ul>
         </div>
-        {this.tabs_getTabView()}
+        <div className="side-panel-tab-content side-panel-section container
+        container-pod container-pod-short container-fluid container-fluid-flush
```
Comment (kennyt) [inline src/js/components/ServiceDetail.js]: these should be indented one deeper

File: src/js/stores/MesosStateStore.js [hunk 9]
```diff
@@ -164,6 +164,37 @@ var MesosStateStore = Store.createStore({
...
+    }
+
+    // Combine framework (if matching framework was found) and filtered
+    // Marathon tasks. This will give you a list of framework tasks including
+    // the scheduler tasks or a list of Marathon application tasks.
+    return frameworks.reduce(function (serviceTasks, framework) {
+      let {tasks=[], completed_tasks={}, name} = framework;
+
+      if (name === mesosServiceName) {
+        return  serviceTasks.concat(tasks,completed_tasks);
```
Comment (kennyt) [inline src/js/stores/MesosStateStore.js]: i'm pretty sure there's an extra space after `return`

File: src/js/components/TaskView.js [hunk 10]
```diff
@@ -133,55 +137,74 @@ class TaskView extends mixin(SaveStateMixin) {
...
-              handleFilterChange={this.handleStatusFilterChange}
-              totalTasksCount={tasks.length}
-              currentStatus={state.filterByStatus}/>
+        <FilterHeadline
+          inverseStyle={inverseStyle}
+          onReset={this.resetFilter}
+          name="Tasks"
+          currentLength={tasks.length}
+          totalLength={totalNumberOfTasks} />
+        <div className="filter-bar">
```
Comment (kennyt) [inline src/js/components/TaskView.js]: there's a new `FilterBar` component that can do all of this for you :)

Comment (orlandohohmeier) [inline src/js/components/TaskView.js]: But it will only replace the div, right? Anyway, let's refactor this in a different PR.

Comment (kennyt) [inline src/js/components/TaskView.js]: it'll replace this div, the `filter-bar-left` div, and the `filter-bar-item` divs

Comment (orlandohohmeier) [inline src/js/components/TaskView.js]: Cool!

File: src/js/components/FilterButtons.js (hunk 1 above)
Comment (orlandohohmeier) [inline src/js/components/FilterButtons.js]: I like the idea, but let's please stick to scope of this PR and extend the `FilterButtons` component in a different PR.

**Comment by Poltergeist** [comment]
Hit the Button 🔘 👍 LGTM works as advertised. :shipit: 🐴
